DashAI/back/api/api_v1/endpoints/plugins.py

```diff
@@ -0,0 +1,284 @@
+@inject
+async def get_plugins(
+    session_factory: Callable[..., ContextManager[Session]] = Provide[
+        Container.db.provided.session
+    ],
+):
+    """Retrieve a list of the stored plugins in the database.
+
+    Parameters
+    ----------
+    session_factory : Callable[..., ContextManager[Session]]
+        A factory that creates a context manager that handles a SQLAlchemy session.
+        The generated session can be used to access and query the database.
+
+    Returns
+    -------
+    List[dict]
+        A list of dictionaries representing the found plugins.
+        Each dictionary contains information about the plugin, including its name,
+        author, tags, description and creation date.
+        If no plugins are found, an empty list will be returned.
+    """
+    with session_factory() as db:
+        try:
+            plugins = db.query(Plugin).all()
+
+        except exc.SQLAlchemyError as e:
+            logger.exception(e)
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Internal database error",
+            ) from e
+
+    return plugins
+
+
+@router.get("/{plugin_id}")
+@inject
+async def get_plugin(
+    plugin_id: int,
+    session_factory: Callable[..., ContextManager[Session]] = Provide[
+        Container.db.provided.session
+    ],
+):
+    """Retrieve the plugin associated with the provided ID.
+
+    Parameters
+    ----------
+    plugin_id : int
+        ID of the plugin to retrieve.
+    session_factory : Callable[..., ContextManager[Session]]
+        A factory that creates a context manager that handles a SQLAlchemy session.
+        The generated session can be used to access and query the database.
+
+    Returns
+    -------
+    Dict
+        A Dict containing the requested plugin details.
+    """
+    with session_factory() as db:
+        try:
+            plugin = db.get(Plugin, plugin_id)
+            if not plugin:
+                raise HTTPException(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    detail="Plugin not found",
+                )
+
+        except exc.SQLAlchemyError as e:
+            logger.exception(e)
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Internal database error",
+            ) from e
+
+    return plugin
+
+
+@inject
+def add_plugin_to_db(
+    raw_plugin: PluginParams,
+    session_factory: Callable[..., ContextManager[Session]] = Provide[
+        Container.db.provided.session
+    ],
+) -> Plugin:
+    """Create a Plugin from a PluginParams instance and store it in the DB.
+
+    Parameters
+    ----------
+    params : List[PluginParams]
+        The new plugins parameters.
+
+    Returns
+    -------
+    List[Plugin]
+        A list with the created plugins.
+    """
+    with session_factory() as db:
+        logging.debug("Storing plugin metadata in database.")
```

Comment:
Use local logger, not the global one (logging)

```suggestion
        logger.debug("Storing plugin metadata in database.")
```